lilypad/_utils.py

```diff
@@ -103,37 +100,41 @@ def get_llm_function_version(
     """Returns the active version for the given function."""
     hash, code = compute_function_hash(fn)
 
-    try:
-        llm_fn_version = lilypad_client.get_llm_function_by_hash(hash)
-    except client.NotFoundError:
-        llm_fn_version = lilypad_client.post_llm_function(
-            function_name=fn.__name__,
-            code=code,
-            version_hash=hash,
-            arg_types=arg_types,
-        )
+    if not synced:
```

Comment:
is this going to affect switching from non-synced to synced for the same function name?

Reply:
the traces if you switch from non-synced to synced will be different under the same function name